x-pack/metricbeat/module/aws/s3_daily_storage/_meta/data.json

```diff
@@ -1,27 +1,28 @@
 {
     "@timestamp": "2017-10-12T08:05:34.853Z",
     "aws": {
-        "s3": {
-            "bucket": {
-                "name": "test-s3-ks-2"
-            }
+        "cloudwatch": {
+            "namespace": "AWS/S3"
+        },
+        "dimensions": {
+            "BucketName": "filebeat-aws-elb-test",
+            "StorageType": "AllStorageTypes"
         },
-        "s3_daily_storage": {
-            "bucket": {
-                "size": {
-                    "bytes": 207372
+        "s3": {
+            "metrics": {
+                "NumberOfObjects": {
+                    "avg": 57828
```

Comment:
Is rename not being applied here?

Reply:
Yeah it's applied when I run metricbeat for testing but not applied when I tried to create the event for data.json. Is there any way I can make the rename processor work with `TestData` function?

Reply:
Oh ok, maybe processors are not initialized on testing, if that is the case we should fix that, but not on this PR. Could you please open an issue about that?

But interestingly in the iis module there is a rename that seems to be applied :thinking: 
https://github.com/elastic/beats/blob/3e5a167809210e9999732d5e50c7e895e10fc389/x-pack/metricbeat/module/iis/website/manifest.yml#L36-L40
https://github.com/elastic/beats/blob/3e5a167809210e9999732d5e50c7e895e10fc389/x-pack/metricbeat/module/iis/website/_meta/data.json#L27

Reply:
Interesting. @narph I saw you generated the data.json file for iis website. Does the field `iis.website.name` generated automatically by `TestData` function? For my case, `TestData` function ignores the rename processor.

Reply:
I am setting the `iis.website.name` using the rename processor inside the manifest.yml file